### ISSUE TITLE
Suggestions for OAGenerateRasterFiles.py

### ISSUE THREAD

**Issue by df511** [pinskylab/OceanAdapt#141]

**Comment by mpinsky** [comment]
Hey @df511, is this the comment from the NOAA researcher? If so, it would help to copy his suggestion in here. Thank you!

**Comment by df511** [comment]
It is, but John has yet to note his suggestions here. I'll follow up now!

**Comment by df511** [comment]
John's changes/comments:

Hi Daniel,

Sorry for the late reply. I started out making a list, but our needs changed a little bit and I ended up needing to make a lot of changes to the original script. The biggest was to make things work in ArcGIS Pro and to enable us to use the Feature Classes generated in our NOAA Geoplatform application. Here is a short list of changes:

1) arcpy.mapping.Layer no longer exists in ArcGIS Pro. I changed that to arcpy.management.MakeFeatureLayer
2) The method used to convert the CSV file to an arcpy table was not consistent across the CSV files. This was mostly how the hualid values changed between regions
3) Some other small things like converting the NA values to Nulls and we needed to create datetime field so that the final layer is time-aware
4) I modified the portion of the script that tests for the smallest value for a year when processing the rasters. The black sea bass of the Gulf of Mexico crashed the script since all WPCUE values where zero 

So, I've been restructuring the script to be used in ArcGIS Pro, to use geodatabases instead of shape files,  separating operations, and adding the ability to filter on regions and species instead of running the script for all regions/species. So, lots of changes and too many for a simple list. I'll share the script with you when I'm done.

Cheers,
John
